gamersnet_backend/routes/posts/updatePost.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+// include our function from the database to add post
+let {updatePostDB, getPost} = require('../../persistence/posts');
+let {verifyUserLoggedIn} = require('../utilities/tokenUtility')
+let {getUserIDFromToken} = require('../../persistence/tokens.js')
+
+
+/**
+ * Helper function: updates specific post. User doesn't need to be logged in.
+ * created this function to test update posts without worrying about tokens being expired.
+ * (Don't wanna go and change expiry time in db everytime I test :/ )
+ * @param {*} request 
+ * @param {*} response 
+ * 
+ * PLEASE DON'T USE THIS IN FRONTEND !! For backend testing only.
+ */
+async function updatePostUnauthorized(request, response) {
+    let body = request.body;
+    let postID = request.query._id; 
+
+    if (postID && body.description && body.gameTimeUTC && body.gameName) {
+        //input type are verified here
+        let numPlayers = parseInt(body.numPlayers);
+        let gameTimeUTC = new Date(body.gameTimeUTC);
+
+        await updatePostDB(postID, body.description, body.gameName, numPlayers, gameTimeUTC, body.duration, body.location);
+
+        response.status(201).send("Updated post successfully.");
+    } else {
+        response.status(400).end();
+    }
+
+}
+
+/**
+ * this function handles the /post/updatePost/ endpoint
+ * Replaces the field values with the given ones in the db
+ * @param {*} request  body should contain the postID(ObjectId or _id), description, gameName, numPlayers, gameTimeUTC, duration, location
+ * @param {*} response 
+ */
+async function updatePost(request, response) {
+    let cookie = request.headers.cookie;
+    let postID = request.query._id;
+    let loggedIn = false;
+
+    if (cookie) {
+        cookie = cookie.split('=')[1];
+        loggedIn = await verifyUserLoggedIn(cookie);
+    }
+
+    if (loggedIn) {
+        let tokenDocument = await getUserIDFromToken(cookie);
+
+        //get logged in user id from token
+        let loggedUserID = tokenDocument.userID;
+        let postUserID = null
+
+        let oldPost = await getPost(postID);
```

Comment:
Won't just passing a post id in the body from the frontend be a simplified and efficient option instead of passing it in the query and then to getPost and then ultimately fetching from getPost?

that's how delete is set up!

Reply:
to update a post we need to fetch it first and when we ask for something from db that is a query.. body contains what we want to POST. So, it wouldn't be logical to put the postID in the body since we are not POSTing it or changing it.. instead we are GETing it from the db. So that's a query. I had it in the body first, but while writing tests I realized it wasn't logical. So i changed it.

Reply:
to add to that, you need to be passing hex form of the id.. whether you add it to the body or the query.